src/Data/Map.purs

```diff
@@ -116,14 +116,24 @@ checkValid tree = length (nub (allHeights tree)) == one
 -- | Lookup a value for the specified key
 lookup :: forall k v. (Ord k) => k -> Map k v -> Maybe v
 lookup _ Leaf = Nothing
-lookup k (Two _ k1 v _) | k == k1 = Just v
-lookup k (Two left k1 _ _) | k < k1 = lookup k left
-lookup k (Two _ _ _ right) = lookup k right
-lookup k (Three _ k1 v1 _ _ _ _) | k == k1 = Just v1
-lookup k (Three _ _ _ _ k2 v2 _) | k == k2 = Just v2
-lookup k (Three left k1 _ _ _ _ _) | k < k1 = lookup k left
-lookup k (Three _ k1 _ mid k2 _ _) | k1 < k && k <= k2 = lookup k mid
-lookup k (Three _ _ _ _ _ _ right) = lookup k right
+lookup k tree =
+  let comp :: k -> k -> Ordering
+      comp = compare
```

Comment:
I wasn't immediately sure why you did this at first, but I get it. Nice. I'll definitely do some work on this one day :wink: